src/NumericalAlgorithms/RootFinding/TOMS748.hpp

```diff
@@ -6,15 +6,465 @@
 
 #pragma once
 
-#include <boost/math/tools/roots.hpp>
 #include <functional>
 #include <limits>
+#include <type_traits>
 
 #include "DataStructures/DataVector.hpp"
 #include "Utilities/ErrorHandling/Error.hpp"
 #include "Utilities/ErrorHandling/Exceptions.hpp"
+#include "Utilities/Simd/Simd.hpp"
 
 namespace RootFinder {
+namespace toms748_detail {
+template <typename T>
+T safe_div(const T& num, const T& denom, const T& r) {
+  if constexpr (std::is_floating_point_v<T>) {
+    using std::abs;
+    if (abs(denom) < static_cast<T>(1)) {
+      if (abs(denom * std::numeric_limits<T>::max()) <= abs(num)) {
+        return r;
+      }
+    }
+    return num / denom;
+  } else {
+    // return num / denom without overflow, return r if overflow would occur.
+    const auto mask0 = fabs(denom) < static_cast<T>(1);
+    // Note: if denom >= 1.0 you get an FPE because of overflow from
+    // `max() * (1. + value)`
+    const auto mask = fabs(simd::select(mask0, denom, static_cast<T>(0)) *
+                           std::numeric_limits<T>::max()) <= fabs(num);
+    return simd::select(mask0 and mask, r, num / denom);
```

Comment:
I have not had an issue, but I'll add a comment with code that could be changed to prevent that. This might have something to do with retiring instructions, etc. but SIMD things like that is something I don't have a complete grasp on. If we have issues, it'll be easy to guard against :)